platform/core/src/feature_file_system.rs

```diff
@@ -0,0 +1,495 @@
+        // TODO: get the path from the platform
+        Ok(env!("CARGO_MANIFEST_DIR").to_string() + "/TestFeatures/")
+    }
+
+    fn feature_name(&self) -> Result<String, String> {
+        // TODO: get the feature name from the platform
+        Ok("Test".to_string())
+    }
+}
+
+fn feature_files_path(config: &impl FeatureFSConfigTrait) -> Result<String, CoreFailure> {
+    let features_path = config
+        .features_path()
+        .map_err(CoreFailure::failed_to_create_feature_files_path)?;
+    let feature_name = config
+        .feature_name()
+        .map_err(CoreFailure::failed_to_create_feature_files_path)?;
+    Ok(features_path + "/" + &feature_name)
+}
+
+type ResourceUrl = String;
+#[allow(dead_code)]
+type ResourceId = String;
+
+#[allow(dead_code)]
+fn resource_url_from_id(id: &ResourceId) -> ResourceUrl {
+    let res_prefix = "polypod://FeatureFiles".to_string();
+    res_prefix + "/" + id
+}
+
+#[allow(dead_code)]
+fn fs_path_from_id(
+    id: &ResourceId,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<String, CoreFailure> {
+    let fs_prefix = feature_files_path(config)?;
+    Ok(fs_prefix + "/" + id)
+}
+
+#[allow(dead_code)]
+fn fs_path_from_resource_url(
+    resource_url: &ResourceUrl,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<String, CoreFailure> {
+    let fs_prefix = feature_files_path(config)?;
+    let res_prefix = "polypod://FeatureFiles".to_string();
+    swap_prefix(resource_url, &res_prefix, &fs_prefix).map_err(|err| {
+        CoreFailure::failed_to_convert_to_fs_path_from_resource_url(resource_url.to_string(), err)
+    })
+}
+
+#[allow(dead_code)]
+fn resource_url_from_fs_path(
+    fs_path: &str,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<String, CoreFailure> {
+    let fs_prefix = feature_files_path(config)?;
+    let res_prefix = "polypod://FeatureFiles/".to_string();
+    swap_prefix(fs_path, &fs_prefix, &res_prefix).map_err(|err| {
+        CoreFailure::failed_to_convert_to_resource_url_from_fs_path(fs_path.to_string(), err)
+    })
+}
+
+fn swap_prefix(string: &str, from: &str, to: &str) -> Result<String, String> {
+    if !string.starts_with(from) {
+        return Err(format!("{} does not start with {}", string, from));
+    }
+    Ok(string.replace(from, to))
+}
+
+#[allow(dead_code)]
+fn make_sure_feature_files_dir_exists(
+    platform_fs: &impl FileSystem,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<(), CoreFailure> {
+    let files_path = feature_files_path(config)?;
+    if platform_fs.exists(&files_path) {
+        platform_fs.create_dir_structure(&files_path)?;
+    }
+    Ok(())
+}
+
+#[allow(dead_code)]
+fn import(
+    url: &Url,
+    dest_resource_url: Option<ResourceUrl>,
+    platform_fs: &impl FileSystem,
+    config: &impl FeatureFSConfigTrait,
+) -> Result<ResourceUrl, CoreFailure> {
+    make_sure_feature_files_dir_exists(platform_fs, config)?;
+
+    let fs_path = match dest_resource_url {
```

Comment:
OK, I see. Really, it's the same code except we're generating the path in advance.